### DISCUSSION TITLE
When repository for CentOS Stream 9/RHEL9?

### DISCUSSION THREAD

**Discussion by hrw** [fluent/fluentd#3704] [Q&A]
CentOS Stream 9 exists for a while, new deployments start, new developments around it.

OpenStack Kolla moves to CS9 but no "td-agent" package for CS9 holds us back ;(

**Comment by cosmo0920** [comment]
There is no docker image for RHEL 9 or CentOS Stream 9 or Rocky Linux 9 that will be binary compatible distribution for RHEL 9.
Our stable Fluentd packaging system heavily depends on Docker containers.
So, we cannot create Fluentd stable distribution packages until that distributions' containers are published.
Sorry for your inconvenience!

**Comment by hrw** [comment]
@cosmo0920 Ever tried quay.io/centos/centos:stream9 image?

**Reply by cosmo0920** [reply]
This image should be provided from official CentOS Stream project?
If not so, we shouldn't use this image for building packages.

**Reply by hrw** [reply]
It is official image. CentOS project migrated to quay.io quite some time ago.

**Reply by cosmo0920** [reply]
Sounds good. Is there any source for this info?

**Reply by hrw** [reply]
Visit https://centos.org/download/ page, switch to "CentOS Stream" and check "Containers" links. 

They all point to https://quay.io/centos/centos:stream9

**Comment by cosmo0920** [comment]
I tried to build CentOS Stream 9 package but CentOS Stream 9 does not provide openssl 1.1.1 version of headers package.
It blocks to build our stable Fluentd packages.

TD-Agent is still staying on Ruby 2.7 and our stable Fluentd distribution, calyptia-fluentd is still on Ruby 3.0.
If we support CentOS Stream 9 w/ our packages, we must bump up using Ruby version to 3.1 or replacing using openssl gem version to 3.0.0 but I'm not sure how to replace Ruby tarball bundled openssl gem version.

**Reply by hrw** [reply]
RHEL9 moved to OpenSSL 3.0 version.

[OpenSSL 3.0 migration guide](https://www.openssl.org/docs/man3.0/man7/migration_guide.html) may be an interesting read.

Ubuntu 22.04 (soon to be released) also moved to OpenSSL 3.0 so you probably need to adapt soon due to having two distributions with newer OpenSSL.

**Reply by ashie** [reply]
Hmm, interesting.
In any case we need to migrate to Ruby 3.1 near future, so that using Ruby 3.1 only for such distribution might be good as a first step of migration.

**Reply by hrw** [reply]
Ubuntu 22.04 is released. And uses Ruby 3.0.2 built against OpenSSL 3.0 one.

So it feels like time to move.

**Comment by hrw** [comment]
RHEL 9 was released. Ruby 3.0.3

**Reply by ashie** [reply]
We'll work on it: https://github.com/fluent/fluent-package-builder/issues/388

**Comment by ashie** [comment]
We deployed an experimental deb package for Ubuntu jammy, which bundles Ruby 3.1:
https://td-agent-package-browser.herokuapp.com/4/ubuntu/jammy/pool/contrib/t/td-agent

**Reply by hrw** [reply]
Now time to do similar for RHEL/CS9...

**Comment by ashie** [comment]
We are trying to build a package for RHEL9 using CentOS Stream 9 container:
https://github.com/fluent/fluent-package-builder/pull/396

I've succeeded to build on my local environment (some minor issues are still remained so that CI is failed though).

BTW do you know whether a container of CentOS Stream 9 for AArch64 exists, and how to pull it? @hrw

**Comment by hrw** [comment]
Container for AArch64 has the same name:

```
11:45 (40s) hrw@jagular:~$ uname -a
Linux jagular 5.17.0-3-arm64 #1 SMP Debian 5.17.11-1 (2022-05-26) aarch64 GNU/Linux

11:45 (0s) hrw@jagular:~$ docker run -it quay.io/centos/centos:stream9
[root@a183b162d0ae /]# cat /etc/os-release
NAME="CentOS Stream"
VERSION="9"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="9"
PLATFORM_ID="platform:el9"
PRETTY_NAME="CentOS Stream 9"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:centos:centos:9"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"
[root@a183b162d0ae /]#
```

**Reply by ashie** [reply]
Could you show the result of `docker inspect quay.io/centos/centos:stream9`?
We are building aarch64 packages on x86_64 host by using qemu on CI environment so that we need to specify aarch64 container explicitly to pull it. AFAIK we need to sha256 digest to do it, `docker inspect` will show it.

**Reply by ashie** [reply]
Now I've found `docker buildx build --platform linux/arm64 ...`, can we use it?

**Reply by hrw** [reply]
I use AArch64 native so no idea.

**Comment by hrw** [comment]
```
12:14 (0s) hrw@jagular:~$ docker pull quay.io/centos/centos:stream9
stream9: Pulling from centos/centos
c346baeaadb7: Pull complete
Digest: sha256:eb9da9af75a9dcbeca9b4acfa223866d02789eae75e00201defcb3446e8e50ff
Status: Downloaded newer image for quay.io/centos/centos:stream9
quay.io/centos/centos:stream9
12:15 (6s) hrw@jagular:~$ docker inspect quay.io/centos/centos:stream9
[
    {
        "Id": "sha256:885771d73d2e73b3a36980d302c7d625162cd45205234ff433fcfb2b6996b269",
        "RepoTags": [
            "quay.io/centos/centos:stream9"
        ],
        "RepoDigests": [
            "quay.io/centos/centos@sha256:eb9da9af75a9dcbeca9b4acfa223866d02789eae75e00201defcb3446e8e50ff"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2022-06-07T00:44:15.109885385Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/bash"
            ],
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "io.buildah.version": "1.19.8",
                "org.label-schema.build-date": "20220607",
                "org.label-schema.license": "GPLv2",
                "org.label-schema.name": "CentOS Stream 9 Base Image",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.vendor": "CentOS"
            }
        },
        "Architecture": "arm64",
        "Os": "linux",
        "Size": 155223258,
        "VirtualSize": 155223258,
        "GraphDriver": {
            "Data": {
                "MergedDir": "/var/lib/docker/overlay2/39e05650ab5745072f4f27e728648e461599d41dbd8bb8536929f48f2da48a63/merged",
                "UpperDir": "/var/lib/docker/overlay2/39e05650ab5745072f4f27e728648e461599d41dbd8bb8536929f48f2da48a63/diff",
                "WorkDir": "/var/lib/docker/overlay2/39e05650ab5745072f4f27e728648e461599d41dbd8bb8536929f48f2da48a63/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:47ac17d71d5602328fed39b965fd238450c2ce83c3183d3a428236e7bdb83666"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
12:15 (1s) hrw@jagular:~$
```

**Comment by CesarTest** [comment]
Hi,
  So, Quaio centos containers
1- Repos not properly installed
2- No Python

How I supposed to rely ansible testing infrastructure on this?

No Python, and can't be installed due to repos issues?

All CI/CD pipelines broken at once.

Test preparation should be quite fix for all pipelines, but minimal setup needed on the container... either Python or repos.
